### PR TITLE
Heretic item animation

### PR DESCRIPTION
**Pull request by Greniza**
## About The Pull Request

The goal of this PR is to add new and interesting animations to the heretic's knives and other equipment.
For example, behold the not only lively but also wriggling Flesh Blade, which is only awake when held by one who knows the arts:
flesh-blade-alt

Or, the fiery Ash Blade:
ash-blade-big

You may notice that its eye does not move around. This is because the eyes are separate overlays, so that animation is easier, and it takes up less space. The flesh blade is an exception to this rule, as its eye translates around as it wriggles. Also of note: the eye is shut when the blade is not being held by a heretic; the above is meant to visualize what the animation set would look like on a specific frame of the eye animation, which is below.
ash-blade-eye-big

Once a heretic gets their blade upgrade (seeking blade, bleeding steel, etc etc) a corresponding overlay or underlay is added while they are holding it.

I'm planning on adding different animations to the blades in hand after the heretic ascends, and, depending on how the specific mechanics of picking up someone else's blade while on a different path from them work (something I'm not quite clear on), I may need to change the knives' code for adding overlays and updating icons.

## Why It's Good For The Game

The purpose behind this PR is to add some interest to the way that heretics are. They are magical, no? I feel like it'd make sense for their knives and other tools to have a little bit of fun animation when they're being held by a heretic.

Furthermore, once a heretic gets a blade upgrade, they can tell which blade upgrade they got by peering at their blade; the individual blade upgrades have unique and distinct animations.

## Changelog
:cl:
imageadd: adds animations and animation framework to the knives and other tools of heretics
/:cl:

### CODE REVIEW

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 /**
- * #Eldritch Knwoledge
+ * #Eldritch Knowledge
```
Comment (Greniza) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:3]: I figured that a single word in a single comment was not worthy of the grammar and formatting tag, but I still ought to fix it.

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 2]
```diff
@@ -91,29 +94,106 @@
...
+			break
+
+	// todo: finish this
+
+/obj/item/melee/sickly_blade/update_appearance()
+	/* This is called when the heretic ascends, gains a blade upgrade, picks up this object, or drops this object.
+	 * The reasoning is that, for one, it's easier, and for two, the blade is an independent object of the heretic;
+	 * Therefore, it is empowered by the heretic and not necessarily innately magical, a magical focus like the wizard's wands.
+	 */
+	var/list/user_knowledge = current_user/datum/antagonist/heretic.researched_knowledge
```
Comment (Fikou) [inline code/modules/antagonists/eldritch_cult/eldritch_items.dm:131]: what?

Comment (Greniza) [reply code/modules/antagonists/eldritch_cult/eldritch_items.dm:131]: This is supposed to set `var/list/user_knowledge` to the list of heretic knowledge that the current wielder (`current_user`) knows, which is a list under `datum/antagonist/heretic` called `researched_knowledge`.

**Comment by YakumoChen** [comment]
Important question: is there any in hands tell for the animation or way to see the full sprite in someone's hand while they have it? Antagonist checks should generally be avoided.

**Comment by Greniza** [comment]
> Important question: is there any in hands tell for the animation or way to see the full sprite in someone's hand while they have it? Antagonist checks should generally be avoided.

In all fairness, in what circumstance have you seen someone wielding a heretic's blade and said to yourself "Oh hey, I think this person might _not_ be a heretic"? Aside from that, antag checks, whether intentional or not, are, I believe, banned on most if not all servers, such as debraining a ling to check whether or not it is a ling (That one not only does not work, but is, I believe, banned when you are doing it exclusively to check for ling status).

All that aside, there is probably a way to override the icon which shows up in chat on examine, and if it is a problem, I will do my best to fix it.

**Pull request closed without merging**